static/js/auth.ts

```diff
@@ -150,90 +127,98 @@ export const auth = {
           ? $('input[name=languages]').filter(':checked').map((_, box) => $(box).val() as string).get()
           : undefined,
       };
 
-      console.log(payload);
-
-      auth.clear_error ();
-      $.ajax ({type: 'POST', url: '/profile', data: JSON.stringify (payload), contentType: 'application/json; charset=utf-8'}).done (function () {
+      $.ajax ({
+        type: 'POST', url: '/profile',
+        data: JSON.stringify (payload),
+        contentType: 'application/json; charset=utf-8'
+      }).done (function () {
         auth.success (auth.texts['profile_updated']);
-        setTimeout (function () {location.reload ()}, 500);
+        setTimeout (function () {location.reload ()}, 1000);
       }).fail (function (response) {
-        if (response.status >= 500) return auth.error (auth.texts['server_error']);
-        auth.error (auth.texts['ajax_error'] + ' ' + response.responseText);
+        show_form_error(response);
       });
     }
 
     if (op === 'change_password') {
-      if (! values.password) return auth.error (auth.texts['please_password'], 'password', '#error-password');
-      if (values.password.length < 6) return auth.error (auth.texts['password_six'], 'password', '#error-password');
-      if (values.password !== values.password_repeat) return auth.error (auth.texts['repeat_match'], 'password_repeat', '#error-password');
-
-      const payload = {old_password: values.old_password, new_password: values.password};
+      const payload = {old_password: values.old_password, password: values.password, password_repeat: values.password_repeat};
 
       auth.clear_error ('#error-password');
-      $.ajax ({type: 'POST', url: '/auth/change_password', data: JSON.stringify (payload), contentType: 'application/json; charset=utf-8'}).done (function () {
-        auth.success (auth.texts['password_updated'], '#success-password');
+      $.ajax ({
+        type: 'POST',
+        url: '/auth/change_password',
+        data: JSON.stringify (payload),
+        contentType: 'application/json; charset=utf-8'
+      }).done (function () {
+        auth.success (auth.texts['password_updated'], '#success_password');
         $ ('#old_password').val ('');
         $ ('#password').val ('');
         $ ('#password_repeat').val ('');
       }).fail (function (response) {
-        if (response.status >= 500) return auth.error (auth.texts['server_error']);
-        if (response.status === 403) auth.error (auth.texts['invalid_password'], null, '#error-password');
-        else                         auth.error (auth.texts['ajax_error'], null, '#error-password');
+        // We have to specify the error element id because the change password page has two forms
+        if (response.status >= 500) {
+          return auth.error (auth.texts['server_error'], null, '#error_password');
+        } else if (response.status == 400) {
+          auth.error (auth.texts[response.responseText], null, '#error_password');
```

Comment:
What if this returns something that's not in the dictionary though?

Reply:
This is a mistake! I had in mind to verify if the value exists, but completely forgot to implement this. We have to add an additional check if the value exists in the dict. The other solution might be as you suggested in the first comment to sent the already translated error message from the back-end.

Reply:
Fixed by no longer be dependent on the error-code. We simply use `auth.error` to show the message from the back-end. We have an `else` to catch any unexpected behavior and then show the `ajax_error`.